### DISCUSSION TITLE
Incorporate inequality gap to a SOC constraint

### DISCUSSION THREAD

**Discussion by nicacevedo** [cvxpy/cvxpy#1949] [Q&A] [answered]
Hello, everyone! First of all, thank you for developing such a useful package for convex optimization in Python. 

I'm currently working on an algorithm that solves multiple SOCP. The main challenge is to reduce the convergence time of each one of these SOCP.  One of the alternatives was to reformulate the SOC constraints of the form

`$$||x||_2 \leq t, \; (x,t) \in \mathbb{R}^{n+1}$$`

and incorporate a gap (**close to zero**) inside the norm, so the problem avoids the apex of the cone, i.e. 

`$$(x,t) \neq 0 \in \mathbb{R}^{n+1}.$$`

The new constraint, with the gap epsilon close to zero, should be then

`$$\sqrt{x'x + \varepsilon} \leq t, \; (x,t) \in \mathbb{R}^{n+1}, \; 0 < \varepsilon \in \mathbb{R},$$`

which satisfies

`$$0 <\sqrt{ \varepsilon} \leq t,$$` 

but  I could not find a way to write this in the `cvxpy.SOC` function or another way to use the `cvxpy` code and make this possible.

Am I missing some documentation/code or there is no way to write such a constraint as a SOC in  `cvxpy`?

**Comment by rileyjmurray** [comment, marked as answer]
Hello! Great to hear from happy users. What you're describing can be accomplished by setting $$y = (x,\sqrt{\epsilon})$$ and then requiring $$(y,t) \in \mathbb{R}^{n+2}$$.

Let me know if that doesn't answer your question :)

**Reply by nicacevedo** [reply]
That's exactly what I needed. I feel so dumb for not seeing it. Thank you very much! :)
